crates/shared/src/current_block.rs

```diff
@@ -27,6 +30,11 @@ pub async fn current_block_stream(
 ) -> Result<watch::Receiver<Block>> {
     let first_block = web3.current_block().await?;
     let first_hash = first_block.hash.ok_or_else(|| anyhow!("missing hash"))?;
+    let first_number = first_block
+        .number
+        .ok_or_else(|| anyhow!("missing number"))?;
+
+    let current_block_number = Arc::new(Mutex::new(first_number.as_u64()));
```

Comment:
ubernit: We can use an `AtomicU64` instead of a `Mutex` if we implement the logic ☝️

Reply:
I started out with an atomic but that didn't correctly allow for reorgs. If we don't have to handle those I also prefer an atomic.

Reply:
Also using an Atomic makes the code way nicer because I don't have to worry about unlocking a `Mutex` before logging stuff.